src/run_part.F90

```diff
@@ -619,6 +474,312 @@ subroutine spec_file_read_parallel_coag_type(file, parallel_coag_type)
 
   end subroutine spec_file_read_parallel_coag_type
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Do a one particle-resolved Monte Carlo simulation timestep.
+#ifdef PMC_USE_CAMP
+  subroutine run_part_timestep(scenario, env_state, aero_data, aero_state, &
+       gas_data, gas_state, run_part_opt, camp_core, photolysis, i_time, &
+       t_start, last_output_time, last_progress_time, i_output, &
+       progress_n_samp, progress_n_coag, progress_n_emit, progress_n_dil_in, &
+       progress_n_dil_out, progress_n_nuc)
+#else
+  subroutine run_part_timestep(scenario, env_state, aero_data, aero_state, &
+       gas_data, gas_state, run_part_opt, i_time, t_start, last_output_time, &
+       last_progress_time, i_output, progress_n_samp, progress_n_coag, &
+       progress_n_emit, progress_n_dil_in, progress_n_dil_out, &
+       progress_n_nuc)
+#endif
```

Comment:
Would it be cleaner (and safer for future modifications) to separate out the CAMP code on a more fine-grained level? This would match what we do below in the argument variable definitions. For example:
```suggestion
  subroutine run_part_timestep(scenario, env_state, aero_data, aero_state, &
       gas_data, gas_state, run_part_opt, &
#ifdef PMC_USE_CAMP
       camp_core, photolysis, &
#endif
       i_time, &
       t_start, last_output_time, last_progress_time, i_output, &
       progress_n_samp, progress_n_coag, progress_n_emit, progress_n_dil_in, &
       progress_n_dil_out, progress_n_nuc)
```